src/Server/Utils/Strategies/RegexResponseStrategy.php

```diff
@@ -67,13 +75,25 @@ private function bodyConditionIsRegex($expectation)
     private function fillWithUrlMatches($expectation, $httpRequest, $responseBody)
     {
         if ($this->urlConditionIsRegex($expectation)) {
-            $responseBody = preg_replace('/\$\{url\.(\d+)\}/', '\$$1', $responseBody);
-
-            return preg_replace(
-                $expectation->getRequest()->getUrl()->getValue(),
-                $responseBody,
-                $httpRequest->getUri()->__toString()
+            $pattern = preg_replace(
+                '/^(.)\^/',
+                '$1',
+                $expectation->getRequest()->getUrl()->getValue()
             );
+            preg_match(
+                $pattern,
+                $httpRequest->getUri()->__toString(),
+                $matches
+            );
+
+            if (isset($matches[1])) {
+                unset($matches[0]);
+                foreach ($matches as $i => $match) {
+                  $replace["\${url.$i}"] = $match;
```

Comment:
Could you please initialize $replace variable? $replace = [];

Reply:
Yes, I can initialize it - but there is no needs in it because if `$mathes[1]` exists by condition the foreach sycle runs at least ones and situation with aninitialized `$replace` variable is not possible.